sdk/nodejs/runtime/resource.ts

```diff
@@ -536,6 +536,16 @@ async function prepareResource(label: string, res: Resource, custom: boolean, re
         if (remote) {
             const componentOpts = <ComponentResourceOptions>opts;
             expandProviders(componentOpts);
+            if (componentOpts.provider !== undefined) {
+                if ((<ProviderResource[]> componentOpts.providers).indexOf(componentOpts.provider) !== -1) {
+                    const pkg = componentOpts.provider.getPackage();
+                    const message = `There is a conflit between the 'provider' field (${pkg}) and a member of the 'providers' map'. `;
+                    const depreciation =  "This will become an error by the end of July 2022. See https://github.com/pulumi/pulumi/issues/8799 for more details";
+                    log.warn(message+depreciation);
+                } else {
+                    (<ProviderResource[]> componentOpts.providers).push(componentOpts.provider);
```

Comment:
Is `componentOpts.providers` guaranteed to be truthy here? I think we need to guard against that.

```suggestion
                    (<ProviderResource[]> componentOpts.providers)?.push(componentOpts.provider);
```

Reply:
We do need to guard against it, but not by bailing out. In case of a `providers === undefined`, we should set `providers = [provider]`. https://github.com/pulumi/pulumi/pull/8923/commits/a27708fc208b12a4f3861813d011465cfd3c1cb1 makes that change.